### PR TITLE
Scenario with more investors

### PR DESCRIPTION
**Pull request by rbharath**
I put together a scenario with more investors. I'm trying to work towards a simulation with the following recipe:

- Creator creates creator block
- Creator direct transfers 100 ETH into reserve
- A datatrust is approved for the market
- 5 investors invest 100 ETH each
- 1 maker gets 1 listing approved in the market 

Working on the test, I'm starting to have to do more refactoring than I thought so I'm making a PR early for feedback.

### CODE REVIEW

**Comment by robrobbins** [comment]
nope. 

you are way over-complicating this. 

just make another context object for your specific test. call it whatever you want -  `investors` maybe. you can do it right in your `deploy_scenario_three_test.go` file:
    // declare your new struct before the vars...
    type Investors struct {
        AuthInvestor1...
        AuthInvestor2...
        ...
    }

    // the `investors` var will be visible to your tests in this package, you can leave the other ones be...
    var context...
    var investors *Investors
    var deployed...
    var deployedError...

    // make yourself a little private getter like how `GetContext` works
    func getInvestors() *Investors {
        ...doStuff...
        return &Investors{
            ...
        }
    }

    func testMain(... {
        context = ...
        investors = getInvestors()
        deployed, deployedError = ...

    }

File: tests/creatorblocktest/scenariothree/deploy_scenario_three_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		authInvestor := test.GetAuthObject()
+		alloc[authInvestor.From] = core.GenesisAccount{Balance: bal}
+		investors = append(investors, authInvestor)
+	}
+	return investors
+}
+
+// variables decalred here have package scope
+var context *test.Ctx
+var investors [](*bind.TransactOpts)
```
Comment (robrobbins) [inline tests/creatorblocktest/scenariothree/deploy_scenario_three_test.go]: why not just a struct like the others and then you can iterate its keys?

File: tests/creatorblocktest/scenariothree/deploy_scenario_three_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	"log"
+	"math/big"
+	"os"
+	"testing"
+	"time"
+)
+
+func getInvestors(bal *big.Int, numInvestors uint64) [](*bind.TransactOpts) {
+	alloc := make(core.GenesisAlloc)
+	investors := make([](*bind.TransactOpts), 0)
```
Comment (robrobbins) [inline tests/creatorblocktest/scenariothree/deploy_scenario_three_test.go]: not sure this is working like you think it is

File: tests/listingtest/deploy_listing_test.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ var deployedError error
 type logr struct{}
 
 func (l *logr) Fatal(a ...interface{}) {
-	log.Fatal(a)
+	log.Fatal(a...)
```
Comment (rbharath) [inline tests/listingtest/deploy_listing_test.go]: The build was failing for some reason until I added this change.

**Comment by rbharath** [comment]
Ok, I think this is ready to merge now. This PR now adds a test for a market where:

- Creator creates market with 1 MKT
- Creator direct transfers 100 ETH to market
- `N` investors invest 100 ETH in the market.

I've tested this with `N` up to 100, and set `N=20` in the code so the test doesn't run too long. The behavior seems reasonable in that the price for each investor increases slightly from the investor previously due to the spread.

**Pull request merged**